src/main/java/edu/ie3/datamodel/io/source/EntitySource.java

```diff
@@ -71,10 +71,10 @@ protected String safeMapGet(Map<String, String> map, String key, String mapName)
    *     optional if no matching entity with the provided uuid can be found
    */
   protected <T extends UniqueEntity> Optional<T> findFirstEntityByUuid(
-      String entityUuid, Collection<T> entities) {
+      UUID entityUuid, Collection<T> entities) {
     return entities.stream()
         .parallel()
-        .filter(uniqueEntity -> uniqueEntity.getUuid().toString().equalsIgnoreCase(entityUuid))
+        .filter(uniqueEntity -> uniqueEntity.getUuid().equals(entityUuid))
```

Comment:
Okay what one should actually do (at least for frequently searched entity sets like nodes) is build a HashMap uuid -> entity, which is way faster. But that's way beyond the scope of this PR.

Reply:
Should I change something in this line or are you creating a new issue for this?

Reply:
Honestly I'd just leave it as it is for now, and I don't think we need an issue for this yet. If we want to optimize for speed at some point, we should measure performance (with profiling or such) anyways.